Source/WebCore/svg/SVGElement.cpp

```diff
@@ -543,6 +543,13 @@ void SVGElement::attributeChanged(const QualifiedName& name, const AtomString& o
 
 void SVGElement::synchronizeAttribute(const QualifiedName& name)
 {
+    // Fast known property access, if not found we fallback to the hashmap property registry.
+    if (auto* property = propertyForAttribute(name)) {
+        if (auto value = property->synchronize())
+            setSynchronizedLazyAttribute(name, AtomString { *value });
+        return;
+    }
+
```

Comment:
This approach is okay. But I think a better approach is to keep accessing the SVG properties through `SVGPropertyOwnerRegistry`. If you add these static methods to `SVGPropertyOwnerRegistry`

```
template<typename T = OwnerType>
static auto fastAnimatedPropertyLookup(T&, const QualifiedName&) -> typename std::enable_if<!std::is_base_of<SVGElement, T>::value, SVGAnimatedProperty*>::type { return nullptr; }

template<typename T = OwnerType>
static auto fastAnimatedPropertyLookup(T& owner, const QualifiedName& attributeName) -> typename std::enable_if<std::is_base_of<SVGElement, T>::value, SVGAnimatedProperty*>::type
{
    return owner.propertyForAttribute(attributeName);
}
```

And if you add this statement to `SVGPropertyOwnerRegistry::synchronize()`

```
if (auto* property = fastAnimatedPropertyLookup(m_owner, attributeName))
    return property->synchronize();
```

Then you will not need to change `SVGElement::synchronizeAttribute()`. Also the same approach can be used with any function which calls `lookupRecursivelyAndApply()`

Reply:
Can you explain why SVGPropertyOwnerRegistry is better?
1. The current `PropertyRegistry` definition is duplicating inheritance relationship. For each SVGElement definition, in addition to class's definition, we need to copy this inheritance information to `SVGPropertyOwnerRegistry`, which is error prone.
2. This field is wasting memory. Each registry is taking 16bytes memory for vtable and Owner reference. Furthermore, it is taking more memory if SVG element is inheriting some SVG element since each class level has this table. For example, SVGRectElement is having 2 PropertyRegistries: SVGRectElement's one and SVGElement's one. So we are using 32bytes for that, but if we just use virtual function, then we can wipe it completely.
3. If we just put them to virtual function of SVGElement itself, it wipes unnecessary indirect accesses through owner references.